src/main/scala/org/constellation/primitives/TransactionGenerator.scala

```diff
@@ -154,7 +157,15 @@ class TransactionGenerator[F[_]: Concurrent: Logger](
     transactionService.put(TransactionCacheData(tx, path = Set(dao.id)))
 
   private def numberOfTransaction(peers: Seq[(Id, PeerData)]): F[Int] =
-    Sync[F].pure { (dao.processingConfig.randomTXPerRoundPerPeer / peers.size + 1) + 1 }
+    roundCounter.getAndIncrement() match {
+      case x if x < transactionsRounds                 => Sync[F].pure(dao.processingConfig.maxTransactionsPerRound)
+      case y if y < (transactionsRounds + emptyRounds) => Sync[F].pure(0)
+      case _                                           => roundCounter.set(0); Sync[F].pure(0)
+    }
+
+  private def emptyRounds = dao.processingConfig.emptyTransactionsRounds
+
+  private def transactionsRounds = dao.processingConfig.amountTransactionsRounds
```

Comment:
Will `dao.processingConfig.amountTransactionsRounds` can be changed later on? Using `def` for such a value will evaluate it each time ("evaluated by name"). If it's an immutable value it's better to use `val` which is evaluated by value (once only) during definition.